space/spacecore/storage/sqlitestorage/space.go

```diff
@@ -0,0 +1,299 @@
+package sqlitestorage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"sync"
+
+	"github.com/anyproto/any-sync/app"
+	"github.com/anyproto/any-sync/commonspace/object/acl/liststorage"
+	"github.com/anyproto/any-sync/commonspace/object/tree/treechangeproto"
+	"github.com/anyproto/any-sync/commonspace/object/tree/treestorage"
+	"github.com/anyproto/any-sync/commonspace/spacestorage"
+	"github.com/anyproto/any-sync/commonspace/spacesyncproto"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	treeTypeTree     = 0
+	treeTypeList     = 1
+	treeTypeSettings = 2
+)
+
+type spaceStorage struct {
+	mu sync.RWMutex
+
+	service *storageService
+
+	spaceId         string
+	spaceSettingsId string
+	aclId           string
+	hash            string
+	oldHash         string
+
+	isDeleted bool
+
+	aclStorage liststorage.ListStorage
+
+	header *spacesyncproto.RawSpaceHeaderWithId
+}
+
+func newSpaceStorage(s *storageService, spaceId string) (spacestorage.SpaceStorage, error) {
+	ss := &spaceStorage{
+		spaceId: spaceId,
+		service: s,
+		header: &spacesyncproto.RawSpaceHeaderWithId{
+			Id: spaceId,
+		},
+	}
+	err := s.stmt.loadSpace.QueryRow(spaceId).Scan(
+		&ss.header.RawHeader,
+		&ss.spaceSettingsId,
+		&ss.aclId,
+		&ss.hash,
+		&ss.oldHash,
+		&ss.isDeleted,
+	)
+	if err != nil {
```

Comment:
Could be 1 if statement
`if err != nil && errors.Is(err, sql.ErrNoRows)`

Reply:
Thanks! There was another bug, I ignored an err in case it was not sql.NoRows